File: cpp/custom_sort.cpp
// https://www.codewars.com/kata/57ea5b0b75ae11d1e800006c/solutions/cpp
//sort array of strings by length
      sort(array.begin(), array.end(),  [](string const &a, string const &b) {return a.length() < b.length();});
// https://www.codewars.com/kata/5a8d2bf60025e9163c0000bc/solutions/cpp
// https://www.codewars.com/kata/53e895e28f9e66a56900011a/solutions/cpp
// https://www.codewars.com/kata/5868b2de442e3fb2bb000119/solutions/cpp
      sort(L.begin(), L.end(),  [](tuple<int, unsigned int, long long> a, tuple<int, unsigned int, long long> b) {
        if (std::get<0>(a)!=std::get<0>(b)) return std::get<0>(a)<std::get<0>(b);
        if (std::get<1>(a)!=std::get<1>(b)) return std::get<1>(a)<std::get<1>(b);
        return std::get<2>(a)<std::get<2>(b);
      });

File: go/custom_sort.go
// https://www.codewars.com/kata/55c6126177c9441a570000cc/solutions/go
// https://www.codewars.com/kata/5868b2de442e3fb2bb000119/solutions/go
  sort.SliceStable(v, func(i, j int) bool { return (v[i][0] < v[j][0]) || (v[i][0] == v[j][0] && v[i][1] < v[j][1]) || (v[i][0] == v[j][0] && v[i][1] == v[j][1] && v[i][2] < v[j][2])})

//2-d string slice
func sorted(arr [][]string) [][]string {
  for _,subArr := range arr { sort.Strings(subArr) }
  sort.SliceStable(arr, func(i, j int) bool {
    for x := range arr[i] {
        if x >= len(arr[j]) { return false }
        if arr[i][x] == arr[j][x] { continue }
        return arr[i][x] < arr[j][x]
    }
    return len(arr[i]) < len(arr[j])
  })
  return arr
}
